spec/statsd_spec.rb

```diff
@@ -162,6 +162,7 @@
 
         before do
           allow(Socket).to receive(:new).and_call_original
+          .with(Socket.pack_sockaddr_un('/tmp/socket')) # fake UDS socket
```

Comment:
This change threw me off, since I saw the `.and_call_original` and thought "wait, how are we faking this if we're calling the original one"?

Testing it out myself, I guess the `.and_call_original` gets ignored when the `.with(...)` is chained. So I would suggest either:

1. Remove the before entirely (since there's a top-level `allow` that mocks out `Socket` anyway)
2. Turn it into an expectation: `expect(Socket).to receive(:new).with(...)` -- this way we actually test that we initialize the `Socket` class correctly.